components/TokenLink/TokenLink.tsx

```diff
@@ -1,19 +1,22 @@
-﻿import React from "react";
+﻿﻿import React from "react";
 import {useRouter} from "next/router";
 import Link from "next/link";
 
-
-type TokenLinkProps = {children: React.ReactNode; href: string}
+type TokenLinkProps = {children: React.ReactNode; href: string; as: string}
 
 const TokenLink: React.FunctionComponent<TokenLinkProps> = ({children, href}: TokenLinkProps) => {
   const router = useRouter();
-  
+    const token = router.query.token;
+  let link = "";
+  if(token === undefined) {
+      link = '/404Error';
+  } else { link = `${href}?token=${token}`}
+
   return (
-      <Link href={`${href}?token=${router.query.token}`}>
+      <Link href={link}>
```

Comment:
Token can't actually be undefined (because otherwise the page wouldn't work) so fine to leave this as it was before.

Reply:
if undefined shows 404?

Reply:
nope - it can't ever be undefined by the time it gets here.